qt/widgets/spectroscopy/src/OutputWidget/OutputNamePresenter.cpp

```diff
@@ -0,0 +1,66 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidQtWidgets/Spectroscopy/OutputWidget/OutputNamePresenter.h"
+
+#include "MantidQtWidgets/Common/WorkspaceUtils.h"
+
+#include <iostream>
+#include <string>
+
+namespace {
+std::string addLabelUnderscore(std::string const &label) { return label.empty() ? "" : "_" + label; }
+} // namespace
+
+namespace MantidQt::CustomInterfaces {
+OutputNamePresenter::OutputNamePresenter(IOutputNameView *view) : m_view(view), m_suffixes(), m_currBasename() {
+  m_view->subscribePresenter(this);
+}
+
+void OutputNamePresenter::setWsSuffixes(std::vector<std::string> const &suffixes) { m_suffixes = suffixes; }
+
+void OutputNamePresenter::setOutputWsBasename(std::string const &outputBasename, std::string const &outputSuffix) {
+  m_view->enableLabelEditor();
+  m_currBasename = outputBasename;
+  m_currOutputSuffix = outputSuffix;
+  handleUpdateOutputLabel();
+}
+
+int OutputNamePresenter::findIndexToInsertLabel(std::string const &outputBasename) {
+  int maxPos = -1;
+  for (auto const &suffix : m_suffixes) {
+    auto pos = static_cast<int>(outputBasename.rfind(suffix));
+    maxPos = pos >= maxPos ? pos : maxPos;
+  }
+  return (m_suffixes.empty() || (maxPos == -1)) ? static_cast<int>(outputBasename.length()) : maxPos;
+}
```

Comment:
The logic in this function could probably go in a Model too because it is pure logic rather than business/GUI logic